src/Database/LSMTree/Internal/Index/Ordinary.hs

```diff
@@ -99,6 +104,18 @@ instance Index IndexOrdinary where
     sizeInPages (IndexOrdinary lastKeys)
         = NumPages $ fromIntegral (length lastKeys)
 
+    headerLBS :: Proxy IndexOrdinary -> LazyByteString
+    headerLBS _ = toLazyByteString        $
+                  word32Host              $
+                  supportedTypeAndVersion
+
+    finalLBS :: NumEntries -> IndexOrdinary -> LazyByteString
+    finalLBS entryCount _ = toLazyByteString $
+                            word64Host       $
+                            fromIntegral     $
+                            unNumEntries     $
+                            entryCount
```

Comment:
I'm a little confused: I see we're defining `headerLBS` and `finalLBS` here, but were they previously not defined elsewhere? Or did ordinary index serialisation previously not include a header and footer?

Reply:
On that note, is incremental serialisation of the index tested at the moment?

Reply:
> I see we’re defining `headerLBS` and `finalLBS` here, but were they previously not defined elsewhere?

Indeed, there hadn’t been an implementation of them yet. Only *deserialization* of headers and footers had been implemented (as part of `fromSBS`). I think this was just because your draft of the desired ordinary index interface that you sent to me in the beginning didn’t contain `headerLBS` and `finalLBS`. 😉

Reply:
> On that note, is incremental serialization of the index tested at the moment?

The actually incremental part of incremental serialization *is* tested: by `prop_incrementalSerialisedKeyListConstructionWorks` in `Test.Database.LSMTree.Internal.Index.Ordinary`. However, construction of the header and footer is not. I’ve created issue #490 for this.